java/google/registry/rde/imports/RdeImportUtils.java

```diff
@@ -62,41 +63,32 @@ public RdeImportUtils(
     this.escrowBucketName = escrowBucketName;
   }
 
-  private <T extends EppResource> boolean importEppResource(final T resource, final String type) {
+  private <T extends EppResource> void importEppResource(final T resource, final String type) {
     @SuppressWarnings("unchecked")
-    final Class<T> resourceClass = (Class<T>) resource.getClass();
-    return ofy.transact(
-        new Work<Boolean>() {
-          @Override
-          public Boolean run() {
-            EppResource existing = ofy.load().key(Key.create(resource)).now();
-            if (existing == null) {
-              ForeignKeyIndex<T> existingForeignKeyIndex =
-                  ForeignKeyIndex.load(
-                      resourceClass, resource.getForeignKey(), clock.nowUtc());
-              // foreign key index should not exist, since existing resource was not found.
-              checkState(
-                  existingForeignKeyIndex == null,
-                  "New %s resource has existing foreign key index; foreignKey=%s, repoId=%s",
-                  type,
-                  resource.getForeignKey(),
-                  resource.getRepoId());
-              ofy.save().entity(resource);
-              ofy.save().entity(ForeignKeyIndex.create(resource, resource.getDeletionTime()));
-              ofy.save().entity(EppResourceIndex.create(Key.create(resource)));
-              logger.infofmt(
-                  "Imported %s resource - ROID=%s, id=%s",
-                  type, resource.getRepoId(), resource.getForeignKey());
-              return true;
-            } else if (!existing.getRepoId().equals(resource.getRepoId())) {
-              logger.warningfmt(
-                  "Existing %s with same id but different ROID. "
-                      + "id=%s, existing ROID=%s, new ROID=%s",
-                  type, resource.getForeignKey(), existing.getRepoId(), resource.getRepoId());
-            }
-            return false;
-          }
-        });
+    Class<T> resourceClass = (Class<T>) resource.getClass();
+    EppResource existing = ofy.load().key(Key.create(resource)).now();
+    if (existing == null) {
+      ForeignKeyIndex<T> existingForeignKeyIndex =
+          ForeignKeyIndex.load(
+              resourceClass, resource.getForeignKey(), clock.nowUtc());
```

Comment:
Does the import happen at time "now"? That's sort of odd, although I dont have a specific better idea. But anyways, you should probably do a direct load without a time, because the FKE might show up as null here if it's past the deletion time, but in this case that's not what you want - you want to check that no FKI ever existed for this ever.